classes/01class/exercises/c01-aws02/ramiray/README.md

```diff
@@ -0,0 +1,97 @@
+1. Created s3 bucket using following command. 
+
+```
+aws s3 mb s3://ramyadevopstest --region ap-southeast-2
+make_bucket: s3://ramyadevopstest
+
+```
+
+2. Uploaded a text file c01-e04.txt to s3.
+
+```
+C:\Users\rramy\Downloads>aws s3 cp c01-e04.txt s3://ramyadevopstest/
+upload: .\c01-e04.txt to s3://ramyadevopstest/c01-e04.txt
+
+```
+
+3. Download the c01-e04.txt file from s3 to the ec2instance which was created as part of  previous exercise 
+
+```
+[root@ip-172-31-45-192 ec2-user]# aws s3 cp s3://ramyadevopstest/c01-e04.txt .
+fatal error: Unable to locate credentials
+
+```
+
+Got the above fatal credential error, so created instance profile  with limited permission to s3 bucket and attached to the instance after which the downlad was successful.
+
+```
+[root@ip-172-31-45-192 ec2-user]# aws s3 cp s3://ramyadevopstest/c01-e04.txt .
+download: s3://ramyadevopstest/c01-e04.txt to ./c01-e04.txt
+```
+
+
+Instance Role details used to access the bucket:
+ 
+```
+aws ec2 describe-instances --instance-ids i-08dbff420f93f73dd --region ap-southeast-2 | grep -i instance-profile
+
+"Arn": "arn:aws:iam::348662207501:instance-profile/Devopsacademyinstancerole"
+```
+Command to create Role, Policy and Instance Profile:
+
+Role: Assume role policy details are provided in the json file.
+
+```
+aws iam create-role --role-name Devopsacademyinstancerole --assume-role-policy-document file://devopsacademytrustpolicy.json
+
+```
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Principal": {
+          "Service": "ec2.amazonaws.com"
+        },
+        "Action": "sts:AssumeRole"
+      }
+    ]
+  }
+
+
+```
+```
+Attaching the s3readonlypolicy to the role: 
+
+```
+aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess --role-name Devopsacademyinstancerole
+
```

Comment:
This AWS predefined policy will grant access to read ALL your buckets.

In terms of security, it would be best to create a policy similar to that one but that only allows access to the bucket you created.

More specifically, the "resources" should be something like `arn:aws:s3:::bucket-name ...` instead of `*`.

Feel free to ask in me here/slack if you don't understand what this is, IMO that is an important concept but we may learn more about it at another class.